kubetest2/pkg/metadata/writer.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metadata
+
+import (
+	"io"
+	"time"
+)
+
+// Writer manages writing out kubetest2 metadata, namely JUnit
+type Writer struct {
+	suite     testSuite
+	start     time.Time
+	runnerOut io.WriteCloser
+	// for faking out time when testing
+	timeNow func() time.Time
```

Comment:
normally we just have a private global fake, but that made testing the output pretty bad